src/System.Private.DeveloperExperience.Console/src/Internal/DeveloperExperience/ConsolePal.Windows.cs

```diff
@@ -3,32 +3,40 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Text;
 
 namespace Internal.DeveloperExperience
 {
     internal static class ConsolePal
     {
-        private static IntPtr s_consoleErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
+        private static IntPtr s_stdErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
 
         internal unsafe static void WriteError(string errorMessage)
         {
-            if (s_consoleErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
+            if (s_stdErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
             {
                 // ensure we have a valid handle before writing to it
                 return; 
             }
 
-            fixed (char *pBuffer = errorMessage)
+            // Ensure that the encoding we use matches the current code page.
+            int currentCodePage = (int)Interop.mincore.GetConsoleOutputCP();
+            Encoding encoding;
+            try
             {
-                int numberOfCharsWritten;
-                Interop.mincore.WriteConsole(s_consoleErrorHandle, (byte*)pBuffer, errorMessage.Length, out numberOfCharsWritten, IntPtr.Zero);
+                encoding = Encoding.GetEncoding(currentCodePage);
```

Comment:
>  am wondering what we are really getting by re-implementing System.Console here. It may be interesting to look into using System.Console as a dependency 

@tarekgh If I'm reading the TFS history right this used to be the case, but got rolled back as part of the move to GitHub. Do you remember why?